src/fight/mod.rs

```diff
@@ -146,14 +112,313 @@ pub fn resolve_fight(regiment_1: regiment::Regiment, regiment_2: regiment::Regim
+        ComputeCase::WORST => ComputeCase::BEST,
+        ComputeCase::MEAN => ComputeCase::MEAN,
+    };
+    let fastest: u8 = computation_tools::find_the_fastest(
+        attacking_regiment.get_model(),
+        defending_regiment.get_model(),
+    );
+    let first_damages: (usize, f64) = compute_case(attacking_regiment, defending_regiment, case);
+    let second_damages: (usize, f64) =
+        compute_case(defending_regiment, attacking_regiment, complementary);
+
+    match fastest {
+        0 => (
+            first_damages.0,
+            second_damages.0,
+            first_damages.1 * second_damages.1,
+        ),
+        1 => {
+            if first_damages.0
+                > defending_regiment
+                    .get_model()
+                    .get_stats()
+                    .get_health_point()
+            {
+                (first_damages.0, 0, first_damages.1)
+            } else {
+                (
+                    first_damages.0,
+                    second_damages.0,
+                    first_damages.1 * second_damages.1,
+                )
+            }
+        }
+        2 => {
+            if second_damages.0
+                > attacking_regiment
+                    .get_model()
+                    .get_stats()
+                    .get_health_point()
+            {
+                (0, second_damages.0, second_damages.1)
+            } else {
+                (
+                    first_damages.0,
+                    second_damages.0,
+                    first_damages.1 * second_damages.1,
+                )
+            }
+        }
+        _ => panic!("Code not supposed to be reached!"),
+    }
+}
+
+/// ## Create a prediction from raw turn computation
+///
+/// ### Parameters
+/// (&regiment::Regiment) first_unit -> The first unit
+///
+/// (&regiment::Regiment) second_unit -> The second unit
+///
+/// (usize, usize, f64) raw -> The raw scenario computation
+///
+/// ### Return
+/// Prediction -> The scenarion represented with the Prediction data structure
+fn turn_result_to_prediction(
+    attacking_regiment: &regiment::Regiment,
+    defending_regiment: &regiment::Regiment,
+    raw: (usize, usize, f64),
+) -> prediction::Prediction {
+    let mut first: regiment::Regiment = attacking_regiment.clone();
+    let mut second: regiment::Regiment = defending_regiment.clone();
+    first.take_damage(raw.1);
+    second.take_damage(raw.0);
+    prediction::Prediction::new(
+        prediction::RegimentResult::new(first, raw.0),
+        prediction::RegimentResult::new(second, raw.1),
+        raw.2,
+    )
+}
+
+/// ## Compute the 3 most important scenario while in melee phase
+///
+/// ### Parameters
+/// (&Unit) first_unit -> The first unit
+///
+/// (&Unit) second_unit -> The second unit
+///
+/// ### Return
+/// (\[Prediction; 3\]) -> \[The more realistic scenario, The best scenario for first unit, The worst scenario for first unit\]
+pub fn create_prediction(
```

Comment:
No need to put `pub`

Reply:
for the moment I'll let it public because it may become the main function for the fight module
Even if it does not, without this function as pulbic, the rest of the code is never use so I have warnings of non used functions
btw I swapped its name with the compute_turn function. It's more accurate like that